resource-types/awssamples-ec2-importkeypair/python/src/awssamples_ec2_importkeypair/handlers.py

```diff
@@ -0,0 +1,566 @@
+        LOG.debug(log_entry)
+    return ProgressEvent.failed(
+        handler_error_code,
+        f"Error: {error_message}",
+    )
+
+
+def _is_callback(
+        callback_context: MutableMapping[str, Any],
+) -> bool:
+    """Logic to determine whether or not a handler invocation is new"""
+    LOG.debug("_is_callback()")
+
+    # If there is a callback context status set, then assume this is a
+    # handler invocation (e.g., Create handler) for a previous request
+    # that is still in progress
+    return callback_context.get('status') == \
+        CALLBACK_STATUS_IN_PROGRESS['status']
+
+
+def _callback_helper(
+        session: Optional[SessionProxy],
+        request: ResourceHandlerRequest,
+        callback_context: MutableMapping[str, Any],
+        model: ResourceModel,
+        is_delete_handler: bool = False,
+) -> ProgressEvent:
+    """Define a callback logic used for resource stabilization"""
+    LOG.debug("_callback_helper()")
+
+    # Call the Read handler to determine status
+    rh = read_handler(
+        session,
+        request,
+        callback_context,
+    )
+    LOG.debug(f"Callback: Read handler status: {rh.status}")
+    # Return success if the Read handler returns success
+    if rh.status == OperationStatus.SUCCESS:
+        return _progress_event_success(
+            model=model,
+            is_delete_handler=False,
+        )
+    elif rh.errorCode:
+        LOG.debug(f"Callback: Read handler error code: {rh.errorCode}")
+        if rh.errorCode == HandlerErrorCode.NotFound and is_delete_handler:
+            # Return a success status if the resource is not found
+            # (thus, assuming it has been deleted).  The Delete
+            # handler's response object must not contain a model:
+            # hence, the logic driven by is_delete_handler set to True
+            # below will not specify a model for ProgressEvent
+            return _progress_event_success(
+                is_delete_handler=True,
+            )
+    # Otherwise, call this handler again by using a callback logic
+    else:
+        return _progress_event_callback(
+            model=model,
+        )
+
+
+def _get_session_client(
+        session: Optional[SessionProxy],
+        service: str,
+) -> type:
+    """Create and return a session client for a given service"""
+    LOG.debug("_get_session_client()")
+
+    if isinstance(
+            session,
+            SessionProxy,
+    ):
+        client = session.client(
+            'ec2',
+        )
+        return client
+    return None
+
+
+def _get_tags_from_model_tags(
+        model_tags: list,
+) -> list:
+    """Create and return a list of key/value tags from model.Tags"""
+    LOG.debug("_get_tags_from_model_tags()")
+
+    tags = [
+        {
+            "Key": model_tag.Key,
+            "Value": model_tag.Value,
+        }
+        for model_tag in model_tags
+    ]
+    return tags
+
+
+def _delete_create_tags_helper(
```

Comment:
Besides the tags from model, we need to merge with the Stack tags.  I think you should have stack tags from `getDesiredResourceTags`